libraries/ts/src/margin/pool/pool.ts

```diff
@@ -1801,6 +1810,56 @@ export class Pool {
     return { riskIndicator, depositRate, borrowRate }
   }
 
+  /// Projects the user's risk level after a swap.
+  projectAfterMarginSwap(
+    marginAccount: MarginAccount, 
+    amount: number,
+    minAmountOut: number | undefined,
+    outputToken: Pool | undefined
+  ): PoolProjection {
+    const defaults = this.getDefaultPoolProjection(marginAccount);
+    if (!minAmountOut || !outputToken) {
+      return defaults;
+    }
+
+    // Prices
+    const inputTokenPrice = this.prices.priceValue.toNumber();
+    const outputTokenPrice = outputToken.prices.priceValue.toNumber();
+
+    // Swap values
+    const inputSwapValue = amount  * inputTokenPrice;
+    const marginAmount = amount - marginAccount.poolPositions[this.symbol].depositBalance.tokens;
+    const marginAmountValue = marginAmount * inputTokenPrice;
+    const minAmountOutValue = minAmountOut * outputTokenPrice;
+
+    // Total Liabilities
+    const totalLiabilities = marginAccount.valuation.liabilities.toNumber() + marginAmountValue;
```

Comment:
```suggestion
    const totalLiabilities = marginAccount.valuation.liabilities.toNumber();
```
This is just the liabilities of the margin account before the swap. The adjustment for a possible borrow is contained in the expression for `riskIndicator` below.